source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-select/index.tsx

```diff
@@ -17,44 +17,57 @@ export interface IOwnProps {
   className?: string;
 }
 
-export type IProps = IOwnProps & WithDataProps & i18nProps;
+type IProps = IOwnProps & WithDataProps & i18nProps;
+
+interface IState {
+  firstTime: boolean;
+}
 
 const translationMap = { 'en-US': 'en', 'es-419': 'es', 'fr-FR': 'fr' };
 
-class LocaleSelect extends React.Component<IProps> {
+class LocaleSelect extends React.Component<IProps, IState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstTime: true,
+    };
+  }
+
   onSelect = async (e, { value }) => {
     e.preventDefault();
 
-    const { i18n } = this.props;
-    const { currentUser, dataStore } = this.props;
+    const { onChange, selected } = this.props;
 
-    i18n.changeLanguage(value);
+    if (value === selected) {
+      return;
+    }
 
-    await update(dataStore, currentUser, {
-      attributes: { locale: value },
-    });
+    onChange(value);
   };
 
   render() {
-    const { currentUser, i18n } = this.props;
+    const { currentUser, i18n, onChange } = this.props;
+    const { firstTime } = this.state;
     const attributes = attributesFor(currentUser) as UserAttributes;
     const userLocale = attributes.locale;
     const { options, language } = i18n;
     const languages = Object.keys(options.resources);
-
-    const selected = userLocale || language;
-
+    const defaultLocale = userLocale || language;
+    if (firstTime) {
+      this.setState({ firstTime: false });
```

Comment:
maybe you want this? https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

but also: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html